src/SFML/Network/Ftp.cpp

```diff
@@ -115,10 +117,24 @@ const char* sfFtpDirectoryResponse_getMessage(const sfFtpDirectoryResponse* ftpD
 
 ////////////////////////////////////////////////////////////
 const char* sfFtpDirectoryResponse_getDirectory(const sfFtpDirectoryResponse* ftpDirectoryResponse)
+{
+    CSFML_CHECK_RETURN(ftpDirectoryResponse, nullptr);
+    return strdup(sf::String(ftpDirectoryResponse->This.getDirectory()).toAnsiString().c_str());
```

Comment:
```suggestion
    return strdup(sf::String(ftpDirectoryResponse->This.getDirectory()).toAnsiString().c_str());
    return strdup(ftpDirectoryResponse->This.getDirectory().string().c_str());
```

No need to use sf::String here. `getDirector()` returns a `std::filesystem::path` which has a `.string()` function that returns an ASCII string from which we can get a `const char*`.